// embeds/dist/types/image.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.render = exports.parse = undefined;

var _slicedToArray = function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; }();

var _dimensions = require('../dimensions');

var _dimensions2 = _interopRequireDefault(_dimensions);

var _virtualElement = require('virtual-element');

var _virtualElement2 = _interopRequireDefault(_virtualElement);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var parse = exports.parse = function parse(_ref) {
  var _ref2 = _slicedToArray(_ref, 1),
      elm = _ref2[0];

  var tagName = elm.tagName.toLowerCase();

  if (tagName === 'img') {
    var _getDimensions = (0, _dimensions2.default)(elm),
        width = _getDimensions.width,
        height = _getDimensions.height;

    return {
      type: 'image',
      src: elm.getAttribute('src'),
      alt: elm.getAttribute('alt') || undefined,
      width: width, height: height
    };
  }

  return null;
};

var render = exports.render = function render(_ref3) {
  var type = _ref3.type,
      src = _ref3.src,
      alt = _ref3.alt,
      width = _ref3.width,
      height = _ref3.height;
  return (0, _virtualElement2.default)('img', { src: src, alt: alt, width: width, height: height });
};
