// embeds/dist/types/video.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.render = exports.parse = undefined;

var _slicedToArray = function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; }();

var _dimensions = require('../dimensions');

var _dimensions2 = _interopRequireDefault(_dimensions);

var _lodash = require('lodash.map');

var _lodash2 = _interopRequireDefault(_lodash);

var _virtualElement = require('virtual-element');

var _virtualElement2 = _interopRequireDefault(_virtualElement);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var type = 'video';

var getSources = function getSources(elm) {
  var sourceElms = elm.getElementsByTagName('source');

  if (sourceElms.length) {
    return (0, _lodash2.default)(sourceElms, function (sourceElm) {
      return {
        src: sourceElm.getAttribute('src'),
        type: sourceElm.getAttribute('type') || null
      };
    });
  }

  return [{
    src: elm.getAttribute('src'),
    type: null
  }];
};

var parse = exports.parse = function parse(_ref) {
  var _ref2 = _slicedToArray(_ref, 1),
      elm = _ref2[0];

  var tagName = elm.tagName.toLowerCase();

  if (tagName === 'video') {
    var _getDimensions = (0, _dimensions2.default)(elm),
        width = _getDimensions.width,
        height = _getDimensions.height;

    var sources = getSources(elm);

    return {
      type: type, sources: sources, width: width, height: height
    };
  }

  return null;
};

var renderSource = function renderSource(_ref3) {
  var src = _ref3.src,
      type = _ref3.type;
  return (0, _virtualElement2.default)('source', { src: src, type: type });
};

var render = exports.render = function render(_ref4) {
  var sources = _ref4.sources,
      width = _ref4.width,
      height = _ref4.height;
  return (0, _virtualElement2.default)(
    'video',
    { width: width, height: height },
    sources.map(renderSource)
  );
};
