// embeds/dist/types/youtube.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.render = exports.parse = undefined;

var _slicedToArray = function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; }();

var _embedlyUrl = require('embedly-url');

var _getYoutubeId = require('get-youtube-id');

var _getYoutubeId2 = _interopRequireDefault(_getYoutubeId);

var _renderIframe = require('../render-iframe');

var _renderIframe2 = _interopRequireDefault(_renderIframe);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var parse = exports.parse = function parse(_ref) {
  var _ref2 = _slicedToArray(_ref, 1),
      elm = _ref2[0];

  var tagName = elm.tagName.toLowerCase();

  if (tagName !== 'iframe') {
    return null;
  }

  var src = elm.getAttribute('src');
  if (!src) {
    return null;
  }

  var embedlyParsed = (0, _embedlyUrl.parse)(src);
  var youtubeId = embedlyParsed ? (0, _getYoutubeId2.default)(embedlyParsed.src) : (0, _getYoutubeId2.default)(src);
  return youtubeId && {
    type: 'youtube',
    youtubeId: youtubeId
  };
};

var render = exports.render = function render(_ref3) {
  var youtubeId = _ref3.youtubeId,
      _ref3$width = _ref3.width,
      width = _ref3$width === undefined ? 640 : _ref3$width,
      _ref3$height = _ref3.height,
      height = _ref3$height === undefined ? 360 : _ref3$height;
  return (0, _renderIframe2.default)({
    src: 'https://www.youtube.com/embed/' + youtubeId,
    width: width, height: height, allowFullscreen: true
  });
};
